### PR TITLE
Link filters

### PR DESCRIPTION
**Pull request by natario1**
Add filtering of deep links when the app is not installed. After this PR, the filter chain will work as follows:

- Remove android shortcuts that could not be opened (because app not installed or invalid)
- **New:** If not installed, remove links that are not web or `android-app://` schemes, since they would not be openable
- **New:** if not installed, remove links that exceed the `not_installed_max_results` value sent by server, if present

If the final number of links is 0 the app result itself is also removed.

### CODE REVIEW

File: BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java [hunk 1]
```diff
@@ -22,10 +28,10 @@
     private static final String APP_NAME_KEY = "app_name";
     private static final String APP_STORE_ID_KEY = "app_store_id";
     private static final String APP_ICON_URL_KEY = "app_icon_url";
-    private static final String APP_SEARCH_DEEP_LINK_KEY = "app_search_deep_link";
     private static final String APP_SCORE_KEY = "score";
     private static final String APP_DEEP_LINKS_KEY = "deep_links";
     private static final String RANKING_HINT_KEY = "ranking_hint";
+    private static final String UNINSTALLED_MAX_RESULTS_KEY = "uninstalled_max_results";
```
Comment (rkhaitan-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java]: small nitpick but can we name this NOT_INSTALLED rather than UNINSTALLED

Comment (rkhaitan-branch) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java]: eh just ignore this

Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java]: Done @rkhaitan-branch !

File: BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java [hunk 2]
```diff
@@ -37,63 +43,93 @@ static BranchSearchResult parse(BranchSearchRequest query, JSONObject object) {
...
+                    if (!isWeb && !isPlayStore) {
+                        linkIterator.remove();
+                    }
                 }
             }
+
+            // Apply the max results constraint
+            // If nothing remains, this app should disappear
+            if (!isInstalled) {
+                int maxResults = resultObj.optInt(NOT_INSTALLED_MAX_RESULTS_KEY, Integer.MAX_VALUE);
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java]: Not sure how the search engine works on the back end but I hope this doesn't expose us to a possibility of showing, say, 1000 results by the the same app (when NOT_INSTALLED_MAX_RESULTS_KEY is missing). If there is a chance of that ever happening, we may want to consider changing the default from Integer.MAX_VALUE to something more reasonable. (Like 10-20?)

Comment (CharlesG-Branch) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchResponseParser.java]: That would only actually happen if the server returned a thousand links for an app, which we have safeguards against.

Also, the previous logic was just to return everything the server provides without any filtering, so at least this won't add any new problems.

**Pull request merged**